logstreams/src/test/java/io/zeebe/logstreams/util/SyncLogStreamBuilder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.logstreams.util;
+
+import io.zeebe.logstreams.log.LogStream;
+import io.zeebe.logstreams.log.LogStreamBuilder;
+import io.zeebe.logstreams.spi.LogStorage;
+import io.zeebe.util.sched.Actor;
+import io.zeebe.util.sched.ActorScheduler;
+import io.zeebe.util.sched.future.ActorFuture;
+import io.zeebe.util.sched.future.CompletableActorFuture;
+
+public final class SyncLogStreamBuilder implements LogStreamBuilder {
+  private final LogStreamBuilder delegate;
+  private ActorScheduler actorScheduler;
+
+  SyncLogStreamBuilder() {
+    this(LogStream.builder());
+  }
+
+  SyncLogStreamBuilder(final LogStreamBuilder delegate) {
+    this.delegate = delegate;
+  }
+
+  @Override
+  public SyncLogStreamBuilder withActorScheduler(final ActorScheduler actorScheduler) {
+    delegate.withActorScheduler(actorScheduler);
+    return this;
+  }
+
+  @Override
+  public SyncLogStreamBuilder withMaxFragmentSize(final int maxFragmentSize) {
+    delegate.withMaxFragmentSize(maxFragmentSize);
+    return this;
+  }
+
+  @Override
+  public SyncLogStreamBuilder withLogStorage(final LogStorage logStorage) {
+    delegate.withLogStorage(logStorage);
+    return this;
+  }
+
+  @Override
+  public SyncLogStreamBuilder withPartitionId(final int partitionId) {
+    delegate.withPartitionId(partitionId);
+    return this;
+  }
+
+  @Override
+  public SyncLogStreamBuilder withLogName(final String logName) {
+    delegate.withLogName(logName);
+    return this;
+  }
+
+  @Override
+  public ActorFuture<LogStream> buildAsync() {
+    return delegate.buildAsync();
+  }
+
+  public SyncLogStream build(final ActorScheduler scheduler) {
```

Comment:
The scheduler is already available right?

Reply:
Except I made the builder class final now :sweat_smile: We can overload the `withActorScheduler` method I guess but its a broken interface, since maybe the method was already called on the original builder and won't be called on the wrapping builder.

e.g.

```java
final LogStreamBuilder originalBuilder = LogStream.builder().withActorScheduler(scheduler);
// ... more stuff
final SyncLogStream logStream = SyncLogStreamBuilder.builder(builder).withActorScheduler(scheduler).build();
```

Of course in our case we know that we don't call it on the original builder, but maybe next year someone sees this and its definitely a confusing API. Or am I overthinking this and nobody will get confused by this? I mean I wouldn't do that in production, but maybe everyone's fine with this for test code?

The other options are to provide getters in the interface, i.e. `LogStreamBuilder#getActorScheduler` so we can get the provided scheduler from the original builder, or to expose the scheduler through the built `LogStream`. In the end, I felt wrapping a builder and having to call `build(ActorScheduler)` was the "least evil" way :sweat_smile: 

But maybe I'm forgetting another way?

Reply:
I would just remove the first `withActorScheduler` call and add a call to `final SyncLogStream logStream = SyncLogStreamBuilder.builder(builder).withActorScheduler(scheduler).build();` and store the scheduler in the SyncLogStream as well.

For me it is more confusing why we have this in the build method again.

Reply:
Ah, I see I left `SyncLogStreamBuilder#withActorScheduler` - I can see how that was confusing for you :sweat_smile: 

I meant to remove it entirely, to keep with the idea that you build a log stream normally (i.e. give it a scheduler) and the `SyncLogStreamBuilder` is just a wrapper around a `LogStreamBuilder` - I wanted to avoid overloading since it normally I would expect to be able to do `LogStreamBuilder#withActorScheduler` and pass that to a `SyncLogStreamBuilder` and just build. To me that felt like a broken API - i.e. you might have to call `withActorScheduler` twice, since the alternative is exposing the actor scheduler to the wrapping instance.

But come to think of it, maybe it's not broken - maybe it's fine, since if you're using `SyncLogStreamBuilder` then its just one of the conditions, you __have__ to specify the actor scheduler, regardless of whether the original builder had it, and it actually makes sense then that they both use the same scheduler.